AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/holder/PreferencesHolder.java

```
@@ -0,0 +1,163 @@
+		FoundPreferenceHolder foundViewHolder = (FoundPreferenceHolder) holder.foundHolders.get(idRefString);
+
+		JBlock block = getAddPreferencesFromResourceBlock();
+		JExpression assignExpression;
+
+		if (foundViewHolder != null) {
+			assignExpression = foundViewHolder.getOrCastRef(preferenceClass);
+		} else {
+			assignExpression = findPreferenceByKey(idRef);
+			if (preferenceClass != null && preferenceClass != classes().PREFERENCE) {
+				assignExpression = cast(preferenceClass, assignExpression);
+			}
+			holder.foundHolders.put(idRefString, new FoundPreferenceHolder(this, preferenceClass, fieldRef, block));
+		}
+
+		block.assign(fieldRef, assignExpression);
+	}
+
+	@Override
+	public FoundPreferenceHolder getFoundPreferenceHolder(JFieldRef idRef, JClass preferenceClass) {
+		String idRefString = codeModelHelper.getIdStringFromIdFieldRef(idRef);
+		FoundPreferenceHolder foundPreferenceHolder = (FoundPreferenceHolder) holder.foundHolders.get(idRefString);
+		if (foundPreferenceHolder == null) {
+			foundPreferenceHolder = createFoundPreferenceAndIfNotNullBlock(idRef, preferenceClass);
+			holder.foundHolders.put(idRefString, foundPreferenceHolder);
+		}
+		return foundPreferenceHolder;
+	}
+
+	private FoundPreferenceHolder createFoundPreferenceAndIfNotNullBlock(JFieldRef idRef, JClass preferenceClass) {
+		JExpression findPreferenceExpression = findPreferenceByKey(idRef);
+		JBlock block = getAddPreferencesFromResourceBlock().block();
+
+		if (preferenceClass == null) {
+			preferenceClass = classes().PREFERENCE;
+		} else if (preferenceClass != classes().PREFERENCE) {
+			findPreferenceExpression = cast(preferenceClass, findPreferenceExpression);
+		}
+
+		JVar preference = block.decl(preferenceClass, "preference", findPreferenceExpression);
+		return new FoundPreferenceHolder(this, preferenceClass, preference, block);
+	}
+
+	@Override
+	public JBlock getPreferenceScreenInitializationBlock() {
+		// not used
```

I prefer this way. Maybe we could throw an 'UnsupportedOperationException'.